ledger/src/blockstore_meta.rs

```diff
@@ -27,11 +27,13 @@ pub struct SlotMeta {
     // The timestamp of the first time a shred was added for this slot
     pub first_shred_timestamp: u64,
     // The index of the shred that is flagged as the last shred for this slot.
+    // None until the shred with LAST_SHRED_IN_SLOT flag is received.
     #[serde(with = "serde_compat")]
     pub last_index: Option<u64>,
     // The slot height of the block this one derives from.
-    // TODO use Option<Slot> instead.
-    pub parent_slot: Slot,
+    // The parent slot of the head of a detached chain of slots is None.
+    #[serde(with = "serde_compat")]
```

Comment:
This backwards compatibility looks good to me. cc @yhchiang-sol and @steviez for future blockstore changes